src/xopen/__init__.py

```diff
@@ -717,6 +721,10 @@ def xopen(
     (parallel gzip) subprocess if possible. See PipedGzipWriter and PipedGzipReader.
 
     When threads = 0, no subprocess is used.
+
+    format overrides the autodetection of input and output formats. This can be
```

Comment:
Being able to override the output format makes sense, but do you see a use case for overriding the input format? I guess it’s fine if only for symmetry ...

Reply:
Not really since we use the magic number to determine the filetype in case of input. (That is how Galaxy's `.dot` files are currently read correctly). 
One good reason to not seperate this between output and input is that it is much simpler, straightforward and least-surprising just to do it this way. And less code (less bugs...).

Reply:
Ok, let’s keep it as it is.